s3stream/src/main/java/com/automq/stream/s3/wal/impl/block/BlockImpl.java

```diff
@@ -93,7 +97,7 @@ public long addRecord(long recordSize, RecordSupplier recordSupplier,
 
         long recordOffset = startOffset + nextOffset;
         recordSuppliers.add(() -> {
-            ByteBuf header = HEADER_POOL.get();
+            ByteBuf header = HEADER_POOL.get().retain();
```

Comment:
When the header is released. Base on ownership principle, when HEADER_POOL#get has already transfer the  ownership of ByteBuf to outside.